lib/Cron/Cleanup.php

```diff
@@ -57,28 +55,24 @@ public function __construct(ITimeFactory $time,
 	}
 
 	protected function run($argument) {
-		$this->logger->debug('Run cleanup job for text sessions');
-		$inactive = $this->sessionService->findAllInactive();
-		/** @var Session $session */
-		foreach ($inactive as $session) {
-			$activeSessions = $this->sessionService->getActiveSessions($session->getDocumentId());
-			if (count($activeSessions) > 0) {
+		$this->logger->debug('Run cleanup job for text documents');
+		$documents = $this->documentService->getAll();
+		foreach ($documents as $document) {
+			$allSessions = $this->sessionService->getAllSessions($document->getId());
+			if (count($allSessions) > 0) {
+				// Do not reset if there are any sessions left
+				// Inactive sessions will get removed further down and will trigger a reset next time
 				continue;
 			}
 
-			$this->documentService->unlock($session->getDocumentId());
-
-			try {
-				$this->logger->debug('Resetting document ' . $session->getDocumentId() . '');
-				$this->documentService->resetDocument($session->getDocumentId());
-				$this->attachmentService->cleanupAttachments($session->getDocumentId());
-				$this->logger->debug('Resetting document ' . $session->getDocumentId() . '');
-			} catch (DocumentHasUnsavedChangesException $e) {
-				$this->logger->info('Document ' . $session->getDocumentId() . ' has not been reset, as it has unsaved changes');
-			} catch (\Throwable $e) {
-				$this->logger->error('Document ' . $session->getDocumentId() . ' has not been reset, as an error occured', ['exception' => $e]);
+			if ($this->documentService->hasUnsavedChanges($document)) {
+				continue;
 			}
+
+			$this->documentService->resetDocument($document->getId());
 		}
+
+		$this->logger->debug('Run cleanup job for text sessions');
 		$removedSessions = $this->sessionService->removeInactiveSessionsWithoutSteps(null);
```

Comment:
Could it be that the document doesn't have sessions left after `removeInactiveSessionsWithoutSteps`? Then we should move this in front of the code that resets documents without sessions, no?

Reply:
I haven't touched those intentionally for now to have an extra buffer of 1 cronjob run to clean up documents where inactive sessions have been removed. Still worth to check in a follow up.